src/python/pants/backend/python/tasks/build_local_python_distributions.py

```diff
@@ -280,30 +266,22 @@ def _create_dist(
             snapshot_fingerprint, is_platform_specific
         )
 
-        cmd = safe_shlex_join(setup_requires_pex.cmdline(setup_py_snapshot_version_argv))
+        cmd = safe_shlex_join(setup_py_runner.cmdline(setup_py_snapshot_version_argv))
         with self.context.new_workunit(
             "setup.py", cmd=cmd, labels=[WorkUnitLabel.TOOL]
         ) as workunit:
-            with pushd(dist_target_dir):
-                result = setup_requires_pex.run(
-                    args=setup_py_snapshot_version_argv,
+            try:
+                setup_py_runner.run_setup_command(
+                    source_dir=Path(dist_target_dir),
```

Comment:
Can this use `PurePath`? `PurePath` is the superclass of `Path` and does not have any methods that access the actual file system, such as not having `Path.exists()` or `Path.read_text()`, but it has everything else. 

It's a safer default, especially with V2-related code, to ensure we aren't doing things we want to avoid like modifying the file system.

Reply:
We could but only giving up a sanity check assertion that `.is_dir()`. I'd prefer to leave as-is especially since SetupPyRunner is only used by 2 v1 tasks. Of the two, only setup-py is ported to v2 and it gets by using v2isms already.